proof/exp.saw

```diff
@@ -70,8 +70,9 @@ me_3 <- admit_cryptol {{ \x k -> Fp.mul (fp_exp x k, x) == fp_exp x (k+1) }};
 me_4 <- admit_cryptol {{ \x n k -> Fp.mul (fp_exp x n, fp_exp x k) == fp_exp x (k+n) }};
 me_5 <- admit_cryptol {{ \x n k -> fp_exp (fp_exp x n) k == fp_exp x (k*n) }};
 me_6 <- admit_cryptol {{ \n -> fp_exp Fp.field_unit n == Fp.field_unit }};
+me_7 <- admit_cryptol {{ \n -> fp_exp Fp.field_zero n == Fp.field_zero }};
```

Comment:
We should define fp_exp for all exponents, not just positive ones.  I did that for fp2_exp,  and that definition says anything to the 0 equals 1.  So that would violate this law, and me_2, me_3.  We need to think about this a bit to make it work conveniently.

Reply:
This is OK.  I revised fp2_exp in a way that makes the rules here all OK.  You might do that here, too

       fp2_exp x e = if e < 0 then Fp_2.div (Fp_2.field_unit, fp2_exp x (-e))
                      | e == 0 /\ x == Fp_2.field_zero then Fp_2.field_zero // needed for the alg. rules
                      | e == 0 then Fp_2.field_unit
                      else Fp_2.mul (x, fp2_exp x (e-1))

This did not break any proofs